prow/repoowners/endpoint.go

```diff
@@ -0,0 +1,77 @@
+package repoowners
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type OwnersServer struct {
+	OwnersClient *Client
+}
+
+type ownersRequest struct {
+	Org        string   `json:"org_name"`
+	Repo       string   `json:"repo_name"`
+	BaseCommit string   `json:"baseCommit"`
+	Paths      []string `json:"filepaths"`
+}
+
+type ownersResponse struct {
+	ConfigMap map[string]*Config `json:"owners,omitempty"`
+}
+
+func (s *OwnersServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Header.Get("Content-Type") != "application/json" {
+		w.WriteHeader(http.StatusNotAcceptable)
+		return
+	}
+
+	rawBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	} else if len(rawBody) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	reqBody := &ownersRequest{}
+
+	if err = json.Unmarshal(rawBody, reqBody); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	ownerIfc, err := s.OwnersClient.LoadRepoOwners(reqBody.Org, reqBody.Repo, reqBody.BaseCommit)
+
+	if err != nil || ownerIfc == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	requestedOwners := &ownersResponse{
```

Comment:
We can make this a one-liner as well given there's only one field to initialise. And remove the empty line below.